data/wicked/check_macvtap.c

```diff
@@ -0,0 +1,140 @@
+// whether the correct answer was received.
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <dirent.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+
+#define PACKET_SIZE 70
+// Expected ARP packet
+// -1 stands for unknown data
+unsigned char arp_packet[PACKET_SIZE] = {
+    0x00, 0x00, 0x00, 0x00, 0x00, // 10 bytes preamble
+    0x00, 0x00, 0x00, 0x00, 0x00,
+    0x0e, 0x0e, 0x0e, 0x0e, 0x0e, 0x0e, // destination MAC address
+    -1, -1, -1, -1, -1, -1,             // source MAC address
+    0x08, 0x06,                         // ARP protocol
+    0x00, 0x01, 0x08, 0x00, 0x06, 0x04, // Ethernet <=> IP
+    0x00, 0x02,                         // "is at" answer
+    -1, -1, -1, -1, -1, -1,             // source MAC address
+    -2, -2, -2, -2,                     // source IP address (replaced with 1st input param )
+    0x0e, 0x0e, 0x0e, 0x0e, 0x0e, 0x0e, // destination MAC address
+    -2, -2, -2, -2,                     // destination IP address ( replaced with 2nd input param)
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // 18 bytes postamble
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+// macvtap file name
+char fn[16];
+
+// Read buffer
+unsigned char buffer[PACKET_SIZE];
+
+// Search for /dev/tapX
+void search_tap()
+{
+    char *hint = "/sys/class/net/macvtap1/ifindex";
+    FILE *fp = fopen(hint, "r");
+    int index;
+
+    if (fp == NULL)
+    {
+        printf("Can't open %s\n", hint);
+        exit(-1);
+    }
+    fscanf(fp, "%d", &index);
+    fclose(fp);
+
+    snprintf(fn, 15, "/dev/tap%d", index);
+    fn[15] = '\0';
+    printf("Type device was found - %s \n", fn);
+}
+
+// Read 70 bytes from the tap device
+void read_packet()
+{
+    int fd;
+
+    fd = open(fn, O_RDONLY);
+    if (fd == -1)
+    {
+        printf("Can't open tap device %s\n", fn);
+        exit(-2);
+    }
+    if (read(fd, &buffer, PACKET_SIZE) < 0)
+    {
+        printf("Error reading %s\n", fn);
+        close(fd);
+        exit(-3);
+    }
+    else
+    {
+        printf("ARP packet catched \n");
+    }
+    close(fd);
+}
+
+void analyze_packet()
+{
+    unsigned char *pbuffer;
+    unsigned char *ppacket;
+    int differ = 0;
+
+    for (ppacket = arp_packet, pbuffer = buffer;
+         ppacket < arp_packet + PACKET_SIZE;
+         ppacket++, pbuffer++)
+    {
+        if (*ppacket != 255)
+        {
+            if (*pbuffer != *ppacket)
+            {
+                printf("arp[%u] buffer[%u] DIFFERS!\n", *ppacket, *pbuffer);
+                differ = 1;
+            }
+            else
+            {
+                printf("arp[%u] buffer[%u] \n", *ppacket, *pbuffer);
+            }
+        }
+    }
+    if (differ == 1)
+    {
+        exit(-4);
+    }
+    else
+    {
+        printf("Success listening to tap device %s, received the expected ARP packet\n", fn);
+    }
```

Comment:
I'd prefer using brackets on the single line as `if`/`else` as it's more compact (even for code which does need brackets at all), but of course your code => your coding style :)

Reply:
totally fine for me , but with one minor correction from my side don't want to do it line by line over and over again . do we have some "tidy" for C which I  could run against this file to cover all your standards regarding code style ? :)